sdk/src/cache/messages/data/scalar/set_batch.rs

```diff
@@ -0,0 +1,158 @@
+/// # let (cache_client, cache_name) = create_doctest_cache_client();
+///
+/// let items = HashMap::from([("k1", "v1"), ("k2", "v2"), ("k3", "v3")]);
+/// let set_batch_request = SetBatchRequest::new(
+///     &cache_name,
+///     items
+/// ).ttl(Duration::from_secs(60));
+///
+/// match cache_client.send_request(set_batch_request).await {
+///     Ok(_) => println!("SetBatchResponse successful"),
+///     Err(e) => if let MomentoErrorCode::CacheNotFoundError = e.error_code {
+///         println!("Cache not found: {}", &cache_name);
+///     } else {
+///         eprintln!("Error setting values in cache {}: {}", &cache_name, e);
+///     }
+/// }
+/// # Ok(())
+/// # })
+/// # }
+/// ```
+pub struct SetBatchRequest<K: IntoBytes + Copy, V: IntoBytes + Copy> {
+    cache_name: String,
+    items: HashMap<K, V>,
+    ttl: Option<Duration>,
+}
+
+impl<K: IntoBytes + Copy, V: IntoBytes + Copy> SetBatchRequest<K, V> {
+    /// Construct a new SetBatchRequest.
+    pub fn new(cache_name: impl Into<String>, items: HashMap<K, V>) -> Self {
+        Self {
+            cache_name: cache_name.into(),
+            items,
+            ttl: None,
+        }
+    }
+
+    /// Set the time-to-live for the batch of items.
+    pub fn ttl(mut self, ttl: impl Into<Option<Duration>>) -> Self {
+        self.ttl = ttl.into();
+        self
+    }
+}
+
+impl<K: IntoBytes + Copy, V: IntoBytes + Copy> MomentoRequest for SetBatchRequest<K, V> {
+    type Response = SetBatchResponse;
+
+    async fn send(self, cache_client: &CacheClient) -> MomentoResult<SetBatchResponse> {
+        // Turn map of items into a vector of keys and vector of SetRequest objects
+        // so we can map keys to the correct SetResponse objects later
+        let mut set_requests: Vec<momento_protos::cache_client::SetRequest> = vec![];
+        let mut set_request_keys: Vec<String> = vec![];
+        for (key, value) in self.items.iter() {
+            let byte_key = key.into_bytes();
+            let set_request = momento_protos::cache_client::SetRequest {
+                cache_key: byte_key.clone(),
+                cache_body: value.into_bytes(),
+                ttl_milliseconds: cache_client.expand_ttl_ms(self.ttl)?,
+            };
+            set_requests.push(set_request);
+            set_request_keys.push(parse_string(byte_key.clone())?);
+        }
+
+        let set_batch_request = prep_request_with_timeout(
+            &self.cache_name,
+            cache_client.deadline_millis(),
+            momento_protos::cache_client::SetBatchRequest {
+                items: set_requests,
+            },
+        )?;
+
+        let mut response_stream = cache_client
+            .next_data_client()
+            .set_batch(set_batch_request)
+            .await?
+            .into_inner();
+
+        // receive stream of get responses
+        let mut responses: HashMap<String, SetResponse> = HashMap::new();
+        let mut set_request_keys_iter = set_request_keys.iter();
```

Comment:
just like in the get batch side, this should be into_iter, and you should not to_string() on line 131 - to avoid extra O(n) memcpys